diff --git a/dwm.c b/dwm.c
index b7a9042..5deb32d 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1085,15 +1085,7 @@ killclient(const Arg *arg)
 {
 	if (!selmon->sel)
 		return;
-	if (!sendevent(selmon->sel, wmatom[WMDelete])) {
-		XGrabServer(dpy);
-		XSetErrorHandler(xerrordummy);
-		XSetCloseDownMode(dpy, DestroyAll);
-		XKillClient(dpy, selmon->sel->win);
-		XSync(dpy, False);
-		XSetErrorHandler(xerror);
-		XUngrabServer(dpy);
-	}
+	XKillClient(dpy, selmon->sel->win);
 }
 
 void
